code/game/machinery/machinery.dm

```diff
@@ -273,19 +273,18 @@ Class Procs:
 // increment the power usage stats for an area
 // defaults to power_channel
 /obj/machinery/proc/use_power(amount, chan = power_channel)
-	var/area/this_area = get_area(src)
 	if(connected_cell && connected_cell.charge > 0)   //If theres a cell directly providing power use it, only for cargo carts at the moment
 		if(connected_cell.charge < amount*0.75)	//Let them squeeze the last bit of power out.
 			connected_cell.charge = 0
 		else
 			connected_cell.use(amount*0.75)
 	else
-		if(!this_area)
-			return 0						// if not, then not powered.
-		if(!powered(chan)) //no point in trying if we don't have power
-			return 0
 
-		this_area.use_power(amount, chan)
+		var/area/power_area = get_area(src)
+		if(power_area && powered(chan, null, power_area)) //no point in trying if we don't have power
+			power_area.use_power(amount, chan)
+		else
+			return 0
```

Comment:
I agree this change is logical but I expect unintended consequences; did you check around the essential places to make sure this didn't brick anything in space, mining, away missions, vaults, etc?

Reply:
I'll test those.